mayan/apps/dynamic_search/search_backends.py

```diff
@@ -195,9 +195,10 @@ def index_related_instance_m2m(
                     }
                     task_kwargs.update(exclude_kwargs)
 
-                    task_index_instance.apply_async(
-                        kwargs=task_kwargs
-                    )
+                    if not setting_search_disable.value:
```

Comment:
Originally tried disabling at the task level (see tasks.py), however, that does not work, as tasks are annotated with the bind decorator:

https://docs.celeryq.dev/en/latest/userguide/tasks.html#bound-tasks

which means that the function itself is executed via a celery task, so basically by the time the code in the task is called, the task has already been generated, sent to the queue, and then pulled from the queue.

As a result, to prevent items from being added to the queue, we need to prevent all calls to dynamic searches task.py

Reply:
" so basically by the time the code in the task is called, the task has already been generated, sent to the queue, and then pulled from the queue." this might mean that we need to dig more into the decorator code.